python_modules/dagster/dagster/_core/snap/execution_plan_snapshot.py

```diff
@@ -93,6 +94,19 @@ def __new__(
             ),
         )
 
+    @property
+    def asset_selection(self) -> AbstractSet[AssetKey]:
+        asset_keys = set()
```

Comment:
my big fear is that there might be some dragon lying out there where this list of assets does not perfectly match with the assets that the job is actually expecting to execute for whatever reason, and it was relatively harmless when we were missing some events to emit, but will not be harmless once we are always using it to subset at the beginning of each run

Reply:
@alangenfeld the reason i added you here is I recall there were cases where the host execution plan snapshot did not perfectly match up with the one in the process (see https://github.com/dagster-io/dagster/pull/22498) and was wondering if this could somehow exacerbate that situation

Reply:
A scenario i can imagine is executing a job that has asset_key annotated outputs on some of its ops but also some bare ops, where we would end up sub-selecting down to just the assets from these changes. 

Also not sure about `asset_selection=None` vs `asset_selection=set()` behavior

Reply:
Can you give an example of what you mean by 'asset_key annotated outputs'?

Reply:
hmmm, would adding a separate planned_assets or targeted_assets field on the run that can we used to efficiently associate runs with assets but is not necc. used for subsetting be one way out of that? seems gross